vin/logger.py

```diff
@@ -0,0 +1,33 @@
+from asyncio.log import logger
+import logging
+import logging.handlers
+import loguru
+import sys
+
+from vin.vin_config import VinConfig
+
+class LoggerConfig:
+    def get(config: VinConfig = None):
+        if not config:
+            config = VinConfig()
+
+        logger = loguru.logger
+        #self.logger = logging.getLogger("")
+        logger = LoggerConfig.__setup_config__(logger, config)
+
+        return logger
+
+    def __setup_config__(logger, config: VinConfig):
+        logger.remove()
+        logger.level(config.log_level)
+
+        logger.info(config.log_to_file)
+        if config.log_to_file:
+            logger.add(config.logfile, rotation = "500 MB")
+
+        if config.log_to_console:
+            logger.add(sys.stderr)
+
+        return logger
+
+logger = LoggerConfig.get()
```

Comment:
Esto creo que sí sería conveniente que lo cambiases, ya que las variables globales de módulo no son una buena práctica, además se inicializa cuando cargas el módulo.

Reply:
Es uno de los puntos que me preocupaba. En principio, el [logger de loguru](https://github.com/Delgan/loguru#ready-to-use-out-of-the-box-without-boilerplate) se carga de esa manera, así que he intentado mantener la misma estructura. 

Podría plantear inicializar un logger diferente. Aunque, por lo que veo en [la documentación](https://loguru.readthedocs.io/en/stable/api/logger.html#loguru._logger.Logger), parece que se debe hacer de esta manera:

> You should not instantiate a `Logger` by yourself, use `from loguru import logger` instead.

Creo que en mi caso no habría problema. Cada vez que se llama a `.get()`, creo que debería crearse un nuevo logger.

Reply:
No deberías crear nuevos loggers, sino tener un objeto logger que compartas usando inyección de código (está en la lista de comprobación actualizada)

Reply:
Según he estado investigando, la forma correcta de crear "singletons" en Python es mediante variables globales de módulos. Hay debate, pero el razonamiento de la mejor respuesta de este [hilo](https://stackoverflow.com/questions/31875/is-there-a-simple-elegant-way-to-define-singletons) me parece convincente. 

De momento, como Loguru invita a que usemos ese tipo de import, voy a dejarlo como está, junto con una pequeña modificación para utilizar correctamente inyección de dependencias.